server/src/main/java/org/apache/cassandra/sidecar/tasks/CdcRawDirectorySpaceCleaner.java

```diff
@@ -182,10 +182,10 @@ protected void routineCleanUp()
     {
         for (InstanceMetadata instanceMetadata : instancesMetadata.instances())
         {
-            List<String> dataDirectories = instanceMetadata.dataDirs();
-            dataDirectories.stream()
-                           .map(dir -> new File(dir, CDC_DIR_NAME))
-                           .forEach(this::cleanUpCdcRawDirectory);
+            if (instanceMetadata.cdcDir() != null)
+            {
+                cleanUpCdcRawDirectory(new File(Optional.ofNullable(instanceMetadata.cdcDir()).get()));
```

Comment:
How about catch all the exception from the method, log a warning, and continue with the next instance?